packages/@lightningjs/ui-components/src/mixins/withThemeStyles/withThemeStyles.mdx

```diff
@@ -0,0 +1,140 @@
+      w: 150,
+      h: 150,
+      rect: true
+    };
+  }
+}
+```
+
+If you want to re-use this `Box` to create a _new_ component called `BlueBox`, you can do that by extending the base class.
+
+```js
+class BlueBox extends Box {
+  static _template() {
+    return {
+      ...super._template(),
+      color: 0xff0000ff
+    };
+  }
+}
+```
+
+This is fine if you want a box that is always blue, but what if you want to change the color of your component without having to extend it and override that single value? Or if you want a box that changes based on certain factors, such as its [mode](?path=/docs/docs-theming-mode--page)? Or what if you want to have multiple color palettes (in LUI, these are called [tones](?path=/docs/docs-theming-tone--page)) that you can switch between depending on if your application is in light or dark mode?
+
+By utilizing `withThemeStyles`, you can tie the color of your box to the tokens used in your theme or a single style object. This can be done in the following ways.
+
+### Defining a Style Object
+
+The first (and suggested) way of handling styles is to wrap your component in the `withThemeStyles` mixin, and then define the `__themeStyle` getter.
+
+```js
+const style = {
+  color: 0xff0000ff
+};
+
+class Box extends withThemeStyles(lng.Component) {
+  static get __componentName() {
+    return 'Box';
+  }
+
+  static get __themeStyle() {
+    return style;
+  }
+
+  static _template() {
+    return {
+      w: 150,
+      h: 150,
+      rect: true
+    };
+  }
+
+  _update() {
+    this.color = this.style.color;
+  }
+}
+```
+
+With this setup, `Box` is able to have its color tied to a style value that exists outside of the `Box` class. This enables us to change the value of color externally, without the need to change any code in the `Box` class. Thus, the class can remain focused on the functional aspects of the component, and all style-related properties can be handled elsewhere.
+
+### Passing In a Style Object
+
+The second (and less preferred option) is to wrap your component in `withThemeStyles` and to pass in your style object as the second argument.
+
+```js
+const style = {
+  color: 0xff0000ff
+};
+
+class Box extends lng.Component {
+  static get __componentName() {
+    return 'Box';
+  }
+
+  static _template() {
+    return {
+      w: 150,
+      h: 150,
+      rect: true
+    };
+  }
+
+  _update() {
+    this.color = this.style.color;
+  }
+}
+
+class NewComponent extends lng.Component {
+  _template() {
+    Box: {
+      type: withThemeStyles(Box, style);
+    }
+  }
+}
+```
+
+This method is usually more appropriate for in-line styling, rather than defining the styles of a new component. This is because you rely on the component already being set up for use with theming, and this method will only apply the given styles to that particular instance of your component. (We advise changes to be made to all instances of a component when possible to ensure proper adherence to your design system.)
```

Comment:
cool just reading this, maybe list some of this before the example above and pull in some of the other recommendations I added